sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -44,7 +45,9 @@ public static ParallelTransferOptions populateAndApplyDefaults(ParallelTransferO
                 : other.getNumBuffers(),
             other.getProgressReceiver(),
             other.getMaxSingleUploadSize() == null ? Integer.valueOf(BlockBlobAsyncClient.MAX_UPLOAD_BLOB_BYTES)
-                : other.getMaxSingleUploadSize());
+                : other.getMaxSingleUploadSize(),
+            // Queues.SMALL_BUFFER_SIZE is the default used by reactor
+            other.getMaxConcurrency() == null ? Integer.valueOf(Queues.SMALL_BUFFER_SIZE) : other.getMaxConcurrency());
```

Comment:
Same comment about what the default should be.

Reply:
https://github.com/reactor/reactor-core/blob/6058a391f614de6213fb85970272fc5b342bd181/reactor-core/src/main/java/reactor/util/concurrent/Queues.java#L88

Looks like that'd be 256 by default. Pretty high.

We limit number of buffers. Shouldn't max concurrency == numBuffers ? Or maxConcurrency be <= numbBuffers with numBuffers being default ?

Reply:
I don't think we should set them to equal because numBuffers isn't used in every parallel operation, so it's not quite the same. 256 does sound high, but again we haven't seen OOM errors before even with this default. On the other hand, it's probably better to err on the side of being a little slower than running out of memory...

Reply:
In v10/11 we defaulted to 5. It was completely arbitrary, but how does that sound here?